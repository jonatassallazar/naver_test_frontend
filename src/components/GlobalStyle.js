import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
    * {
    box-sizing: border-box;
    outline: 0;
  }
  body {
    margin: 0;
    padding: 0;
  }
  html, body {
    min-height: 100vh;
    min-width: 100vw;
  }
  html {
    font-size: 10px;
    font-family: Montserrat;
  }
`;

export default GlobalStyle;
