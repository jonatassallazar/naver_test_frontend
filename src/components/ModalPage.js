import React from 'react';
import styled from 'styled-components';
import { CloseIcon } from './Icon';
import Button from './Button';

const Modal = styled.div`
  background: ${({ theme }) => theme.colors.bgHover};
  width: 100%;
  height: 100%;
  position: fixed;
  display: flex;
  align-items: center;
  justify-content: center;
  left: 0;
  top: 0;
  z-index: 5;
`;

Modal.Content = styled.div`
  width: 50%;
  background: ${({ theme }) => theme.colors.bg};
  display: flex;
  flex-direction: column;
  position: relative;
  padding: ${({ theme }) => theme.spacing.medium} ${({ theme }) => theme.spacing.large};
`;

Modal.Title = styled.h2`
  font-weight: 600;
  font-size: 2.4rem;
  line-height: 3.6rem;
  margin: 0 0 ${({ theme }) => theme.spacing.small} 0;
`;

Modal.Description = styled.p`
  font-weight: normal;
  font-size: 1.6rem;
  line-height: 3.6rem;
  margin: 0 0 ${({ theme }) => theme.spacing.small} 0;
`;

Modal.Close = styled.div`
  position: absolute;
  right: 2rem;
  top: 2rem;

  svg {
    width: 1.4rem;
  }
`;

Modal.Buttons = styled.div`
  width: 100%;
  display: flex;
  justify-content: flex-end;
  flex-direction: row;
  flex-wrap: wrap;

  button {
    margin: 0 ${({ theme }) => theme.spacing.small};
    width: fit-content;
  }
`;

const ModalPage = ({
  title,
  description,
  buttonPrimary,
  buttonSecondary,
  closeButton = true,
}) => (
  <Modal>
    <Modal.Content>
      {closeButton && (
        <Modal.Close>
          <CloseIcon />
        </Modal.Close>
      )}
      {title && <Modal.Title>{title}</Modal.Title>}
      {description && <Modal.Description>{description}</Modal.Description>}
      {buttonPrimary || buttonSecondary ? (
        <Modal.Buttons>
          {buttonSecondary && (
            <Button.Outlined>{buttonSecondary}</Button.Outlined>
          )}
          {buttonPrimary && <Button>{buttonPrimary}</Button>}
        </Modal.Buttons>
      ) : undefined}
    </Modal.Content>
  </Modal>
);

export default ModalPage;
