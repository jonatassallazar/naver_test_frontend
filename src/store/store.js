import { createContext } from 'react';
import { deleteNaver } from '../axios/instance';

export const initialState = {
  navers: [],
  modalInfo: {
    title: '',
    description: '',
    buttonPrimary: undefined,
    buttonSecondary: undefined,
    closeButton: true,
  },
  showModal: false,
  updated: false,
};

export default createContext(initialState);

export const reducer = (state, action) => {
  switch (action.type) {
    case 'ADD_NAVER':
      return {
        ...state,
        modalInfo: {
          ...initialState.modalInfo,
          title: 'Naver criado',
          description: 'Naver criado com sucesso!',
        },
        showModal: true,
        updated: false,
      };
    case 'EDIT_NAVER':
      return {
        ...state,
        modalInfo: {
          ...initialState.modalInfo,
          title: 'Naver atualizado',
          description: 'Naver atualizado com sucesso!',
        },
        showModal: true,
        updated: false,
      };
    case 'SET_NAVER':
      return {
        ...state,
        navers: action.data,
        updated: true,
      };
    case 'CONFIRMATION_DELETE_NAVER':
      return {
        ...state,
        modalInfo: {
          ...initialState.modalInfo,
          title: 'Excluir Naver',
          description: 'Tem certeza que deseja excluir este Naver?',
          buttonPrimary: 'Excluir',
          actionButtonPrimary: () => {
            const result = deleteNaver(action.id);
            result
              .then(() => {
                action.dispatch({ type: 'DELETE_NAVER' });
              })
              .catch((err) => {
                console.log(err);
              });
          },
          buttonSecondary: 'Cancelar',
          actionButtonSecondary: () => {
            action.dispatch({ type: 'CLOSE_MODAL' })
          },
          closeButton: false,
        },
        showModal: true,
        updated: false,
      };
    case 'DELETE_NAVER':
      return {
        ...state,
        modalInfo: {
          ...initialState.modalInfo,
          title: 'Naver excluído',
          description: 'Naver excluído com sucesso!',
        },
        showModal: true,
        updated: false,
      };

    case 'CLOSE_MODAL':
      return {
        ...state,
        showModal: false,
        modalInfo: {
          ...initialState.modalInfo
        }
      };
    default:
      return state;
  }
};
