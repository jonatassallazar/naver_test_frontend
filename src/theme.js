const theme = {
  colors: {
    primary: '#212121',
    secondary: '',
    bg: '#ffffff',
    bgHover: 'rgba(0, 0, 0, 0.5)',
    text: '#212121',
    textContrast: '#ffffff'
  },
  spacing: {
    tiny: '0.8rem',
    small: '1.4rem',
    medium: '2.4rem',
    large: '3.2rem',
    huge: '6rem'
  }
};

export default theme;