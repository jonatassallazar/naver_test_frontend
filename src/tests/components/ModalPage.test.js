import ModalPage from '../../components/ModalPage';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Switch, BrowserRouter } from 'react-router-dom';
import theme from '../../theme';
import navers from '../fixtures/fixtures';
import store from '../../store/store';
import GlobalStyle from '../../components/GlobalStyle';
import 'jest-styled-components';

const state = {
  navers,
  modalInfo: {
    title: '',
    description: '',
    buttonPrimary: undefined,
    buttonSecondary: undefined,
    closeButton: true,
  },
  showModal: false,
  updated: false,
};
const dispatch = jest.fn();

test('should match ModalPage snapshot with no data', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      <store.Provider value={{ state, dispatch }}>
        <BrowserRouter>
          <Switch>
            <ModalPage />
          </Switch>
        </BrowserRouter>
      </store.Provider>
    </ThemeProvider>,
  );

  expect(container).toMatchSnapshot();
});

test('should match ModalPage snapshot with title and description', () => {
  const state = {
    modalInfo: {
      title: 'Mock Title',
      description: 'Mock Description',
      buttonPrimary: undefined,
      buttonSecondary: undefined,
      closeButton: true,
    },
  };

  const { container } = render(
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      <store.Provider value={{ state, dispatch }}>
        <BrowserRouter>
          <Switch>
            <ModalPage />
          </Switch>
        </BrowserRouter>
      </store.Provider>
    </ThemeProvider>,
  );

  expect(container).toMatchSnapshot();
});

test('should match ModalPage snapshot with title and description and buttons', () => {
  const state = {
    modalInfo: {
      title: 'Mock Title',
      description: 'Mock Description',
      buttonPrimary: 'Mock button',
      buttonSecondary: 'Mock button',
      closeButton: false,
    },
  };

  const { container } = render(
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      <store.Provider value={{ state, dispatch }}>
        <BrowserRouter>
          <Switch>
            <ModalPage />
          </Switch>
        </BrowserRouter>
      </store.Provider>
    </ThemeProvider>,
  );

  expect(container).toMatchSnapshot();
});