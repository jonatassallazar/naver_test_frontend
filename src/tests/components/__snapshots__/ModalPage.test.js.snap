// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match ModalPage snapshot with no data 1`] = `
.c2 {
  width: 1.8rem;
  height: 1.8rem;
}

.c3 {
  background: rgba(0,0,0,0.5);
  width: 100%;
  height: 100%;
  position: fixed;
  left: 0;
  top: 0;
  z-index: 2;
}

.c0 {
  width: 60%;
  background: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2.4rem 3.2rem;
  z-index: 6;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1 {
  position: absolute;
  right: 2rem;
  top: 2rem;
  cursor: pointer;
}

.c1 svg {
  width: 1.4rem;
}

@media (max-width:768px) {
  .c0 {
    text-align: center;
    width: 90%;
    padding: 3.2rem 1.4rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <svg
        class=" c2"
        version="1.1"
        viewBox="0 0 14 14"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <path
          d="M14 1.41L12.59 0L7 5.59L1.41 0L0 1.41L5.59 7L0 12.59L1.41 14L7 8.41L12.59 14L14 12.59L8.41 7L14 1.41Z"
          fill="currentColor"
        />
      </svg>
    </div>
    
    
  </div>
  <div
    class="c3"
  />
</div>
`;

exports[`should match ModalPage snapshot with title and description 1`] = `
.c2 {
  width: 1.8rem;
  height: 1.8rem;
}

.c5 {
  background: rgba(0,0,0,0.5);
  width: 100%;
  height: 100%;
  position: fixed;
  left: 0;
  top: 0;
  z-index: 2;
}

.c0 {
  width: 60%;
  background: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2.4rem 3.2rem;
  z-index: 6;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c3 {
  font-weight: 600;
  font-size: 2.4rem;
  line-height: 3.6rem;
  margin: 0 0 1.4rem 0;
}

.c4 {
  font-weight: normal;
  font-size: 1.6rem;
  line-height: 3.6rem;
  margin: 0 0 1.4rem 0;
}

.c1 {
  position: absolute;
  right: 2rem;
  top: 2rem;
  cursor: pointer;
}

.c1 svg {
  width: 1.4rem;
}

@media (max-width:768px) {
  .c0 {
    text-align: center;
    width: 90%;
    padding: 3.2rem 1.4rem;
  }
}

@media (max-width:768px) {
  .c4 {
    line-height: 1;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <svg
        class=" c2"
        version="1.1"
        viewBox="0 0 14 14"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <path
          d="M14 1.41L12.59 0L7 5.59L1.41 0L0 1.41L5.59 7L0 12.59L1.41 14L7 8.41L12.59 14L14 12.59L8.41 7L14 1.41Z"
          fill="currentColor"
        />
      </svg>
    </div>
    <h2
      class="c3"
    >
      Mock Title
    </h2>
    <p
      class="c4"
    >
      Mock Description
    </p>
  </div>
  <div
    class="c5"
  />
</div>
`;

exports[`should match ModalPage snapshot with title and description and buttons 1`] = `
.c4 {
  background: #212121;
  color: #ffffff;
  font-style: normal;
  font-weight: 600;
  font-size: 1.4rem;
  line-height: 2.4rem;
  text-align: center;
  border: none;
  padding: 1rem 5rem;
  width: 100%;
  font-family: inherit;
  cursor: pointer;
}

.c5 {
  background: none;
  color: #212121;
  border: 1px solid #212121;
}

.c6 {
  background: rgba(0,0,0,0.5);
  width: 100%;
  height: 100%;
  position: fixed;
  left: 0;
  top: 0;
  z-index: 2;
}

.c0 {
  width: 60%;
  background: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2.4rem 3.2rem;
  z-index: 6;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1 {
  font-weight: 600;
  font-size: 2.4rem;
  line-height: 3.6rem;
  margin: 0 0 1.4rem 0;
}

.c2 {
  font-weight: normal;
  font-size: 1.6rem;
  line-height: 3.6rem;
  margin: 0 0 1.4rem 0;
}

.c3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c3 button {
  margin: 0 1.4rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

@media (max-width:768px) {
  .c0 {
    text-align: center;
    width: 90%;
    padding: 3.2rem 1.4rem;
  }
}

@media (max-width:768px) {
  .c2 {
    line-height: 1;
  }
}

@media (max-width:768px) {
  .c3 button {
    width: 100%;
    margin: 0.4rem;
  }
}

<div>
  <div
    class="c0"
  >
    <h2
      class="c1"
    >
      Mock Title
    </h2>
    <p
      class="c2"
    >
      Mock Description
    </p>
    <div
      class="c3"
    >
      <button
        class="c4 c5"
      >
        Mock button
      </button>
      <button
        class="c4"
      >
        Mock button
      </button>
    </div>
  </div>
  <div
    class="c6"
  />
</div>
`;
